rmf_traffic_editor/gui/delete_dialog.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "delete_dialog.h"
+#include <algorithm>
+#include <QtWidgets>
+using std::vector;
+using std::string;
+
+
+DeleteDialog::DeleteDialog(
+  QWidget* parent,
+  Building& building,
+  int level_idx,
+  int selected_vertex_idx)
+: QDialog(parent)
+{
+
+
+  setWindowTitle("Warning");
+  QVBoxLayout* warning_message_vbox_layout = new QVBoxLayout;
+
+  // check if vertex is a lift cabin
+  std::vector<std::string> lift_levels;
+  auto level = building.levels[level_idx];
+  const auto v = level.vertices[selected_vertex_idx];
+  auto it = v.params.find("lift_cabin");
+  if ((it != v.params.end()))
+  {
+    auto lift = building.get_lift(v.lift_cabin());
+    for (const auto& level : building.levels)
+    {
+      if (level.elevation <= lift.highest_elevation &&
+        level.elevation >= lift.lowest_elevation)
+      {
+        lift_levels.push_back(level.name);
+      }
+    }
+    std::string s = lift_levels[0];
+    for (uint i = 1; i < lift_levels.size(); i++)
+    {
+      s = s + ", " + lift_levels[i];
+    }
+    warning_message_vbox_layout->addWidget(new QLabel(
+        QString("This item is a lift cabin and is used in %1.\n "
+        "Are you sure you want to delete this item?")
+        .arg(QString::fromStdString(s))));
+  }
+  else
+  {
+    std::string s;
+
+    auto num_used_edges = level.edges_with_vertex(selected_vertex_idx).size();
+    std::unordered_map<std::string, int> edge_types;
```

Comment:
Nit, when printing to the user it's nice to have a deterministic order, I would suggest using `std::map` here so they are always in alphabetical order, rather than a non deterministic hash map based one.

Reply:
Changed to std::map instead as suggested.